valhalla/midgard/util.h

```diff
@@ -27,13 +27,14 @@
 namespace valhalla {
 namespace midgard {
 
-// Holds a range plus a default value for that range
+// Holds a named option, its range plus a default value for that range
 template <class T> struct ranged_default_t {
   T min, def, max;
 
   // Returns the value snapped to the default if outside of the range
-  T operator()(const T& value) const {
+  T operator()(const T& value, bool& clamped) const {
     if (value < min || value > max) {
+      clamped = true;
```

Comment:
it's basically following the suggestion here: https://github.com/valhalla/valhalla/pull/3833#discussion_r1030396723